test/promise.js

```diff
@@ -187,4 +187,71 @@ describe('Promise', function () {
       assert(prom instanceof SubClass);
     });
   });
-}); 
\ No newline at end of file
+  describe('Promise.all', function() {
+    it('throws on non arrays', function () {
+      var actual = [];
+      var expected = [
```

Comment:
Can we put these values in separate tests? I prefer tests to be as simple as possible. It's really hard to debug a failure when one test is testing multiple things as once.

```js
var originalImmediate;
beforeEach(function() {
   // Make promise behave synchonously
   originalImmediate = Promise._immediateFn;
   Promise._immediateFn = function(fn) {
     fn();
  };
});
afterEach(function() {
   Promise._immediateFn = originalImmediate;
});
it('error on undefined implicit', function() {
   var error;
   return Promise.all().catch(function(result) {
     error = result;
   });
   assert.ok(error instanceof Error);
});

// ...
```

Reply:
@taylorhakes done, ive separated those tests, kept async nature of the tests though. Imho its easier to follow the code when it uses normal Promise API, instead of hacking internal `_immediateFn` in `beforeEach`/`afterEach` hooks. If you want to have this changed further - let me know.